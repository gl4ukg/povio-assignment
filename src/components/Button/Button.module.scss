.button {
    padding: 23px 100px;
    border-radius: 3px;
    background: #fff;
    color: #DF9186;
    font-family: 'Montserrat', sans-serif;
    font-weight: 500;
    border: 0;
    cursor: pointer;
    display: flex;
    align-items: center;
    justify-content: center;
    height: 50px;
    &__isRounded {
        border-radius: 50px;
    }
    &__isColored {
        background: linear-gradient(270deg, #ECBCB3 0%, var(--main) 100%);
        color: #fff;
    }
    &__isBllack {
        background: #00000080!important;
        color: #fff!important;
    }
    &__isSmall {
        padding: 23px;
        height: 40px;
        align-items: center;
        display: flex;
    }
}