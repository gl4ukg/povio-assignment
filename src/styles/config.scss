:root {
    --white: #fff;
    --main: #EAA79E;
   
    --light-grey: #949EA0;
    --grey: #334144;

}