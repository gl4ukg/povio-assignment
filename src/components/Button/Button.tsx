import styles from "./Button.module.scss"
import classNames from "classnames"

interface Props {
    text: string,
    isRounded?: boolean,
    isColored?: boolean,
    isSmall?: boolean,
    isBlack?: boolean,
    isDisabled?: boolean,
    isSubmit?: boolean,
    onClick?: () => void,
    className?: string,
    isIcon?: string
}

const Button:React.FC<Props> = (props: Props) => {
    return (
        <button 
            disabled={props.isDisabled}
            type={props.isSubmit ? 'submit': 'button'}
            className={classNames(styles.button, {
                [styles.button__isColored]: props.isColored,
                [styles.button__isRounded]: props.isRounded,
                [styles.button__isSmall]: props.isSmall,
                [styles.button__isBlack]: props.isBlack,
                [props.className as string]: props.className,
                [styles.button__isIcon]: props.isIcon
            })}
            onClick={props.onClick}
            >
            {props.isIcon && <img src={props.isIcon} alt={props.isIcon} />}
            {props.text}
        </button>
    )
}

export default Button