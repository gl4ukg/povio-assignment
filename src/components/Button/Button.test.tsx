import { render, screen } from '@testing-library/react';
import styles from "./Button.module.scss"
import Button from './Button';

describe('button', () => {
    it('should render with text prop correctly', () => {
        const { container } = render(<Button 
            text='Hello I am testing'
        />);

        expect(container).toBeInTheDocument();
        expect(container).toHaveTextContent('Hello I am testing');
    })

    it('should have rounded class when rounded prop is true', () => {
        const { container } = render(<Button 
            text='Hello I am testing'
            isRounded
        />);
        
        expect(container).toBeInTheDocument();
        expect(container.firstChild).toHaveClass(styles.button__isRounded);
    })

    it('should have colored class when colored prop is true', () => {
        const { container } = render(<Button 
            text='Hello I am testing'
            isColored
        />);
        
        expect(container).toBeInTheDocument();
        expect(container.firstChild).toHaveClass(styles.button__isColored);
    })

    it('should have small class when small prop is true', () => {
        const { container } = render(<Button 
            text='Hello I am testing'
            isSmall
        />);
        
        expect(container).toBeInTheDocument();
        expect(container.firstChild).toHaveClass(styles.button__isSmall);
    })

    it('should have black class when black prop is true', () => {
        const { container } = render(<Button 
            text='Hello I am testing'
            isBlack
        />);
        
        expect(container).toBeInTheDocument();
        expect(container.firstChild).toHaveClass(styles.button__isBlack);
    })

    it('should have isIcon class when isIcon prop is present', () => {
        const { container } = render(<Button 
            text='Hello I am testing'
            isIcon='photo'
        />);
        
        expect(container).toBeInTheDocument();
        expect(container.firstChild).toHaveClass(styles.button__isIcon);
    })

    it('should have img element when isIcon prop is present with isIcon as src and alt', () => {
        const { container } = render(<Button 
            text='Hello I am testing'
            isIcon='photo'
        />);
        const imgElement = container.getElementsByTagName('img')[0];


        expect(container).toBeInTheDocument();
        expect(imgElement).toHaveAttribute('src', 'photo');
        expect(imgElement).toHaveAttribute('alt', 'photo');
    })

    it('should have given class name when class name prop is present', () => {
        const { container } = render(<Button 
            text='Hello I am testing'
            className='test_class_name'
        />);
        
        expect(container).toBeInTheDocument();
        expect(container.firstChild).toHaveClass('test_class_name');
    })
});


describe('button snapshots', () => {
    it('should render correctly with text value', () => {
        const { container } = render(<Button text='Testing'/>)

        expect(container.firstChild).toMatchSnapshot()
    });

    it('should render correctly with isIcon value and have img tag', () => {
        const { container } = render(<Button 
                text='Testing'
                isIcon='test'
            />
        )
        
        expect(container.firstChild).toMatchSnapshot()
    });
})