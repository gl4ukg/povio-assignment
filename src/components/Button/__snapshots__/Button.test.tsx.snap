// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button snapshots should render correctly with isIcon value and have img tag 1`] = `
<button
  class="button button__isIcon"
  type="button"
>
  <img
    alt="test"
    src="test"
  />
  Testing
</button>
`;

exports[`button snapshots should render correctly with text value 1`] = `
<button
  class="button"
  type="button"
>
  Testing
</button>
`;
